### ISSUE TITLE
HEAD redirection problem

### ISSUE THREAD

**Issue by tnt-dev** [benoitc/hackney#168]
Current version doesn't handle redirection of HEAD requests correct.

**Comment by benoitc** [comment]
It should be:

https://github.com/benoitc/hackney/blob/master/src/hackney_client/hackney.erl#L726

How can I reproduce your issue?

**Comment by edgurgel** [comment]
Just tested with master:

```
4> hackney:head("http://httparrot.herokuapp.com/redirect/1", [], "", [{follow_redirect, true}]).
{ok,200,
    [{<<"Connection">>,<<"keep-alive">>},
     {<<"Server">>,<<"Cowboy">>},
     {<<"Date">>,<<"Wed, 25 Feb 2015 06:13:43 GMT">>},
     {<<"Content-Length">>,<<"407">>},
     {<<"Content-Type">>,<<"application/json">>},
     {<<"Via">>,<<"1.1 vegur">>}]}
5> hackney:head("http://httparrot.herokuapp.com/redirect/1", [], "", [{follow_redirect, false}]).
{ok,301,
    [{<<"Connection">>,<<"keep-alive">>},
     {<<"Server">>,<<"Cowboy">>},
     {<<"Date">>,<<"Wed, 25 Feb 2015 06:14:07 GMT">>},
     {<<"Content-Length">>,<<"0">>},
     {<<"Content-Type">>,<<"text/html">>},
     {<<"Location">>,<<"http://httparrot.herokuapp.com/get">>},
     {<<"Via">>,<<"1.1 vegur">>}]}
```

**Comment by tnt-dev** [comment]
Oh, I needed to be more detailed, sorry.

HEAD request return reply here https://github.com/benoitc/hackney/blob/master/src/hackney_client/hackney.erl#L888
but in https://github.com/benoitc/hackney/blob/master/src/hackney_client/hackney.erl#L788 there is no case for it and it returns as Error and I think it's not normal.

Redirecting works, but unexpected way.

**Comment by benoitc** [comment]
@edgurgel  thanks for the test :)

Hrm not sure I totally understand the issue. It is still returing `{ok, Status, Headers}` right ? In that case maybe it should be renamed `Else` instead of `Error`?

**Comment by benoitc** [comment]
fixed in 928106c75b3778156cd8ca3c69a832efe8489f4c
